tests/test_rdmacm.py

```diff
@@ -1,32 +1,48 @@
 # SPDX-License-Identifier: (GPL-2.0 OR Linux-OpenIB)
 # Copyright (c) 2019 Mellanox Technologies, Inc. All rights reserved. See COPYING file
 
-from tests.rdmacm_utils import sync_traffic, async_traffic, \
-    async_traffic_with_ext_qp
+from tests.rdmacm_utils import  CMSyncConnection, CMAsyncConnection
 from pyverbs.pyverbs_error import PyverbsError
 from tests.base import RDMATestCase
 import multiprocessing as mp
 import pyverbs.device as d
 import subprocess
 import unittest
 import json
+import os
+
 
 NUM_OF_PROCESSES = 2
 
 
 class CMTestCase(RDMATestCase):
     def setUp(self):
+        if self.dev_name is not None:
```

Comment:
#747 change breaks rdmacm tests on IB devices in two possible ways:
1. When there's multiple devices/ports but there's just one active device/port, but the latter doesn't have a net interface IP the test would fail on IndexError (since we're trying to get the IP when there's none).
2. In RDMATestCase base class it randomizes which device and args to use, so in case we have more than one device in an active state but just one of them has net IP, it might not be selected, hence some rdmacm tests would run properly but other get skipped (since they could use another device with no net interface)

@Kamalheib currently, I've reverted the change in this PR. But you're welcome to give any feedback/suggestion.

Reply:
@EdwardSro: I think the test needs to be fixed by skip the run if the selected interface doesn't have a configured ip address, as there is no guarantee that each interface has a configured ip address.

Reply:
This was the behavior before;
If the user explicitly passes the device to be used, and it doesn't have a net interface with configured IP then the test is skipped.
But in case the user doesn't pass the device arg, CMTestCases search and select a device with a configured IP interface, if none were found then the test is skipped.
If I understood you correctly, relying on RDMATestCase chosen device can result in skipping some rdmacm tests while they might have the ability to run on other device(s)

Reply:
@EdwardSro: Well I think the bug exists from day zero because there is no check for "IndexError" when the "self.dev_name" is set...

And yes, The CMTestCase needs to relay on the selected device (even when that is done randomly), probably we need to make sure that the selected device has a configured IP address in RDMATestCase to avoid doing what was done in test_rdmacm.py in new tests.

Reply:
> And yes, The CMTestCase needs to relay on the selected device (even when that is done randomly), probably we need to make sure that the selected device has a configured IP address in RDMATestCase to avoid doing what was done in test_rdmacm.py in new tests.

I've done so, will upload a new commit for this in this series.